Could not resolve all files for configuration ':journeyapps_react-native-quick-sqlite:androidJdkImage'

I have created a fresh expo project using ```npx create-expo-app app -t``` with ```Navigation (TypeScript)``` template and installed and configured powersync as per the instruction at https://github.com/powersync-ja/powersync-js/blob/main/packages/powersync-sdk-react-native/README.md

### When I build the project to android, I get the following error
```
> Task :journeyapps_react-native-quick-sqlite:compileDebugJavaWithJavac FAILED

> Task :expo:compileDebugKotlin
w: file:///data/myapp.me/app/node_modules/expo/android/src/main/java/expo/modules/ReactActivityDelegateWrapper.kt:139:50 No cast needed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':journeyapps_react-native-quick-sqlite:compileDebugJavaWithJavac'.
> Could not resolve all files for configuration ':journeyapps_react-native-quick-sqlite:androidJdkImage'.
   > Failed to transform core-for-system-modules.jar to match attributes {artifactType=_internal_android_jdk_image, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for JdkImageTransform: /home/user/Android/Sdk/platforms/android-34/core-for-system-modules.jar.
         > Error while executing process /usr/lib/jvm/java-21-openjdk/bin/jlink with arguments {--module-path /home/user/.gradle/caches/transforms-3/5fc9c09a9df6137601206ecef23d2934/transformed/output/temp/jmod --add-modules java.base --output /home/user/.gradle/caches/transforms-3/5fc9c09a9df6137601206ecef23d2934/transformed/output/jdkImage --disable-plugin system-modules}

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.3/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 16s
170 actionable tasks: 3 executed, 167 up-to-date
Error: /data/myapp.me/app/android/gradlew exited with non-zero code: 1
Error: /data/myapp.me/app/android/gradlew exited with non-zero code: 1
    at ChildProcess.completionListener (/data/myapp.me/app/node_modules/@expo/spawn-async/build/spawnAsync.js:52:23)
    at Object.onceWrapper (node:events:633:26)
    at ChildProcess.emit (node:events:518:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5)
    ...
    at Object.spawnAsync [as default] (/data/myapp.me/app/node_modules/@expo/spawn-async/build/spawnAsync.js:17:21)
    at spawnGradleAsync (/data/myapp.me/app/node_modules/@expo/cli/build/src/start/platforms/android/gradle.js:72:46)
    at Object.assembleAsync (/data/myapp.me/app/node_modules/@expo/cli/build/src/start/platforms/android/gradle.js:52:18)
    at runAndroidAsync (/data/myapp.me/app/node_modules/@expo/cli/build/src/run/android/runAndroidAsync.js:36:24)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

### package.json
```json
{
  "name": "myapp.me",
  "main": "expo-router/entry",
  "version": "1.0.0",
  "scripts": {
    "start": "expo start",
    "android": "expo run:android",
    "ios": "expo run:ios",
    "web": "expo start --web",
    "test": "jest --watchAll"
  },
  "jest": {
    "preset": "jest-expo"
  },
  "dependencies": {
    "@azure/core-asynciterator-polyfill": "^1.0.2",
    "@expo/vector-icons": "^14.0.0",
    "@journeyapps/powersync-sdk-react-native": "^1.2.0",
    "@journeyapps/react-native-quick-sqlite": "^1.1.1",
    "@react-navigation/native": "^6.0.2",
    "base-64": "^1.0.0",
    "expo": "~50.0.6",
    "expo-font": "~11.10.2",
    "expo-linking": "~6.2.2",
    "expo-router": "~3.4.7",
    "expo-splash-screen": "~0.26.4",
    "expo-status-bar": "~1.11.1",
    "expo-system-ui": "~2.9.3",
    "expo-web-browser": "~12.8.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-native": "0.73.4",
    "react-native-fetch-api": "^3.0.0",
    "react-native-polyfill-globals": "^3.1.0",
    "react-native-safe-area-context": "4.8.2",
    "react-native-screens": "~3.29.0",
    "react-native-url-polyfill": "^2.0.0",
    "react-native-web": "~0.19.6",
    "text-encoding": "^0.7.0",
    "web-streams-polyfill": "^3.3.2",
    // "react-native-get-random-values": "^1.10.0" // Already installed as a peer-dependency (currently explicit installing raises an error related to dependency conflict)
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@babel/plugin-transform-async-generator-functions": "^7.23.9",
    "@types/react": "~18.2.45",
    "jest": "^29.2.1",
    "jest-expo": "~50.0.2",
    "react-test-renderer": "18.2.0",
    "typescript": "^5.1.3"
  },
  "private": true
}
```

### Developement Environment
```
OS: Manjaro (kernel 6.6)
Android sdk: 34 (emulator)
openjdk 21 2023-09-19
OpenJDK Runtime Environment (build 21+35)
OpenJDK 64-Bit Server VM (build 21+35, mixed mode, sharing)
```

PS: Before installing / configuring powersync, android build worked fine.

### Same error

```
> Task :journeyapps_react-native-quick-sqlite:compileDebugJavaWithJavac FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':journeyapps_react-native-quick-sqlite:compileDebugJavaWithJavac'.
> Could not resolve all files for configuration ':journeyapps_react-native-quick-sqlite:androidJdkImage'.
   > Failed to transform core-for-system-modules.jar to match attributes {artifactType=_internal_android_jdk_image, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for JdkImageTransform: /home/user/Android/Sdk/platforms/android-34/core-for-system-modules.jar.
         > Error while executing process /usr/lib/jvm/java-21-openjdk/bin/jlink with arguments {--module-path /home/user/.gradle/caches/transforms-3/5fc9c09a9df6137601206ecef23d2934/transformed/output/temp/jmod --add-modules java.base --output /home/user/.gradle/caches/transforms-3/5fc9c09a9df6137601206ecef23d2934/transformed/output/jdkImage --disable-plugin system-modules}

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.3/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3m 7s
205 actionable tasks: 73 executed, 132 from cache
Error: /data/powersync-js/demos/react-native-supabase-todolist/android/gradlew exited with non-zero code: 1
Error: /data/powersync-js/demos/react-native-supabase-todolist/android/gradlew exited with non-zero code: 1
    at ChildProcess.completionListener (/data/powersync-js/demos/react-native-supabase-todolist/node_modules/@expo/cli/node_modules/@expo/spawn-async/build/spawnAsync.js:52:23)
    at Object.onceWrapper (node:events:633:26)
    at ChildProcess.emit (node:events:518:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5)
    ...
    at Object.spawnAsync [as default] (/data/powersync-js/demos/react-native-supabase-todolist/node_modules/@expo/cli/node_modules/@expo/spawn-async/build/spawnAsync.js:17:21)
    at spawnGradleAsync (/data/powersync-js/demos/react-native-supabase-todolist/node_modules/@expo/cli/build/src/start/platforms/android/gradle.js:72:46)
    at Object.assembleAsync (/data/powersync-js/demos/react-native-supabase-todolist/node_modules/@expo/cli/build/src/start/platforms/android/gradle.js:52:18)
    at runAndroidAsync (/data/powersync-js/demos/react-native-supabase-todolist/node_modules/@expo/cli/build/src/run/android/runAndroidAsync.js:36:24)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

--- a/demos/react-native-supabase-todolist/android/build.gradle
+++ b/demos/react-native-supabase-todolist/android/build.gradle
@@ -15,7 +15,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath('com.android.tools.build:gradle')
+        classpath('com.android.tools.build:gradle:8.2.2')
         classpath('com.facebook.react:react-native-gradle-plugin')
     }
 }

--- a/demos/react-native-supabase-todolist/android/gradle/wrapper/gradle-wrapper.properties
+++ b/demos/react-native-supabase-todolist/android/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-all.zip
 networkTimeout=10000
 validateDistributionUrl=true
 zipStoreBase=GRADLE_USER_HOME

